django/contrib/auth/hashers.py

```diff
@@ -512,6 +512,88 @@ class BCryptPasswordHasher(BCryptSHA256PasswordHasher):
     digest = None
 
 
+class ScryptPasswordHasher(BasePasswordHasher):
+    """
+    Secure password hashing using the Scrypt algorithm
+
+    The result is a 64 byte binary string.
+    """
+    algorithm = 'scrypt'
+    block_size = 8
+    maxmem = 0
+    parallelism = 1
+    work_factor = 2 ** 14
+
+    def encode(self, password, salt, n=None, r=None):
+        assert password is not None
+        assert salt and '$' not in salt
+        n = n or self.work_factor
+        r = r or self.block_size
+        hash_ = hashlib.scrypt(
+            password.encode(),
+            salt=salt.encode(),
+            n=n,
+            r=r,
+            p=self.parallelism,
+            maxmem=self.maxmem,
+            dklen=64,
+        )
+        hash_ = base64.b64encode(hash_).decode('ascii').strip()
+        return '%s$%d$%s$%s$%d$%d' % (
+            self.algorithm, n, salt, hash_, r, self.parallelism,
+        )
+
+    def decode(self, encoded):
+        algorithm, work_factor, salt, hash_, block_size, parallelism = encoded.split('$', 6)
```

Comment:
Is there a good reason to order the data like this? I'd personally expect the hash to be at the end, so it could include a `$` .

Reply:
Good point, updated :+1: